_posts/2015-09-26-using-vim-to-drive-tddd.md

```diff
@@ -0,0 +1,98 @@
+---
+layout: post
+title:  "Using Vim to Drive TDD"
+date:   2015-09-26 09:43:02
+categories: vim
+---
+
+A lot of recent developments in my programming career have been inspired by [thoughtbot](https://thoughtbot.com/), this is the result of following their blog and listening to their podcasts. Two of the major changes I have been working on are, adopting Vim and testing driving my development. Fortunately these activities are not mutually exclusive and I am able to develop my abilities in both simultaneously.
+
+## Specs Should Run in Vim
+
+In order to actually follow TDD I need to be running my tests constantly. If I want to run my tests constantly I need it to be easy -  preferably not requiring any sort of context switch. Running specs in Vim would help me to avoid the context switch but could it be easy?
+
+Thanks to the thoughtbot team and their Vim plugin, [vim-rspec](https://github.com/thoughtbot/vim-rspec), it can be!
+
+This plugin adds functions for running all specs, the entire current spec file, the nearest spec(s), and to re-run the most recent spec. These functions can be [mapped to leader commands](https://github.com/thoughtbot/vim-rspec#key-mappings) to allow you to run them quickly.
+
+<img src='/images/run_spec_default.gif'></img>
+
+To really get into the TDD flow I run specs basically anytime I am in normal mode for more than a few seconds, with this plugin was essentially spamming my new leader commands. I find that unless I am quickly editing a recent change I generally "sit" in normal mode while I am thinking of my next move. This thinking time lends it self to running specs.
```

Comment:
I think the intention of this was to demonstrate how often I run specs now but I'm not sure this paragraph is doing this well.